homeassistant/components/sensor/oasa.py

```diff
@@ -0,0 +1,202 @@
+    vol.Required(CONF_BUS): cv.string,
+    vol.Required(CONF_LAT): cv.string,
+    vol.Required(CONF_LON): cv.string,
+    vol.Optional(CONF_DIRECTION): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the OASA public transport sensor."""
+    name = config.get(CONF_NAME)
+    bus = config.get(CONF_BUS)
+    lat = config.get(CONF_LAT)
+    lon = config.get(CONF_LON)
+    direction = config.get(CONF_DIRECTION)
+
+    data = PublicTransportData(bus, lat, lon, direction)
+    add_devices([OASAPublicTransportSensor(data, name)], True)
+
+
+class OASAPublicTransportSensor(Entity):
+    """Implementation of an Dublin public transport sensor."""
+
+    def __init__(self, data, name):
+        """Initialize the sensor."""
+        self.data = data
+        self._name = name
+        self._times = self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._times is not None:
+            return {
+                ATTR_BUS: self._times[0][ATTR_BUS],
+                ATTR_ROUTE: self._times[0][ATTR_ROUTE],
+                ATTR_STOP: self._times[0][ATTR_STOP],
+                ATTR_DIRECTION: self._times[0][ATTR_DIRECTION],
+                ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return 'min'
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the latest data from opendata.ch and update the states."""
+        self.data.update()
+        self._times = self.data.info
+        try:
+            self._state = self._times[0][ATTR_ARRIVAL]
+        except TypeError:
+            pass
+
+
+def getCompassDirection(degrees):
+    degrees = int(degrees)
+    direction = "n/a"
+    direction2 = "n/a"
+    if 315 <= degrees <= 45:
+        direction = "N"
+    elif 45 <= degrees <= 135:
+        direction = "E"
+    elif 135 <= degrees <= 225:
+        direction = "S"
+    elif 225 <= degrees <= 315:
+        direction = "W"
+
+    if 0 <= degrees <= 90:
+        direction2 = "NE"
+    elif 90 <= degrees <= 180:
+        direction2 = "SE"
+    elif 180 <= degrees <= 270:
+        direction2 = "SW"
+    elif 270 <= degrees <= 360:
+        direction2 = "NW"
+
+    return direction, direction2
+
+
+class PublicTransportData(object):
+    """The Class for handling the data retrieval."""
+
+    def __init__(self, bus, lat, lon, direction):
+        """Initialize the data object."""
+        bus = ast.literal_eval(bus)
```

Comment:
Why do you need ast?

Isn't bus always int / string ?

Reply:
You are right. I is unnecessary.
When I started writing this component, I thought that all buses were integers so this part of the code stayed like this.